Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -119,18 +133,45 @@ public bool TryGetXComponentCorrelationId(string instrumentationKey, out string
                             {
                                 this.GenerateCorrelationIdAndAddToDictionary(instrumentationKey, appId.Result);
                             }
+                            catch (AggregateException ex)
+                            {
+                                ex.Flatten();
+                                if (ex.InnerExceptions != null && ex.InnerExceptions.Count > 0 && ex.InnerExceptions[0] != null)
+                                {
+                                    this.RegisterFailure(instrumentationKey, ex.InnerExceptions[0]);
+                                }
+                                else
+                                {
+                                    this.RegisterFailure(instrumentationKey, ex);
+                                }
+                            }
                             catch (Exception ex)
                             {
-                                CrossComponentCorrelationEventSource.Log.FetchAppIdFailed(this.GetExceptionDetailString(ex));
+                                this.RegisterFailure(instrumentationKey, ex);
```

Comment:
It seems like Commit d03ab986 made by @iusafaro  with the following comments have corrupted this in develop. The version of the file before that change in develop is the correct version. Please address that merge.

```
Move Sergey's fix to remove BCL to develop (#416)

* remove bcl references from projects

* Switch versions back to 2.3.0

* Switch version back to 2.3.0

* Add catch exception

* Catch AggregateException

```